crates/db_views_moderator/src/mod_add_community_view.rs

```diff
@@ -44,14 +45,21 @@ impl ModAddCommunityView {
       query = query.filter(mod_add_community::community_id.eq(community_id));
     };
 
+    if let Some(other_person_id) = other_person_id {
+      query = query.filter(person_alias_1::id.eq(other_person_id));
+    };
+
     let (limit, offset) = limit_and_offset(page, limit);
 
-    let res = query
+    let mut res = query
       .limit(limit)
       .offset(offset)
       .order_by(mod_add_community::when_.desc())
       .load::<ModAddCommunityViewTuple>(conn)?;
 
+    res.iter_mut().for_each(|item| {
+      item.1.name = "Moderator".to_string();
+    });
```

Comment:
I think this is how you hide the mod name? You cant hardcode and English language string like this. Instead, you should change the name field to an Option, and ideally exclude that field from the query. If thats too complicated, you can simply make a helper function instead of copy-pasting the same code everywhere.

Reply:
Also i think that the mod name has to be visible to admins and other mods of the community. For example, in case an account gets hacked, they need to be able to block it. There is a function `is_mod_or_admin()` somewhere that you can use. And when you hide the name, i believe there is also a field userid which needs to be set to None.

Reply:
Mark this as a draft PR, and just ping us whenever you need help, or would like us to take a look.

Also this file is the one to look at for this:

https://github.com/LemmyNet/lemmy/blob/main/crates/db_views_moderator/src/structs.rs#L50

See all those structs like `ModBanView`? The `moderator: PersonSafe` in those needs to be a `moderator: Option<PersonSafe>`, and only be set if you are a moderator, or that hide_mod_names config is false.